pkg/frontend/adminactions/resources_list_test.go

```diff
@@ -24,111 +24,129 @@ import (
-				resources.EXPECT().ListByResourceGroup(gomock.Any(), "test-cluster", "", "", nil).Return([]mgmtfeatures.GenericResourceExpanded{
-					{
-						Name: to.StringPtr("vm-1"),
-						ID:   to.StringPtr("/subscriptions/id"),
-						Type: to.StringPtr("Microsoft.Compute/virtualMachines"),
-					},
-					{
-						Name: to.StringPtr("storage"),
-						ID:   to.StringPtr("/subscriptions/id"),
-						Type: to.StringPtr("Microsoft.Storage/storageAccounts"),
-					},
-				}, nil)
-
-				virtualNetworks.EXPECT().Get(gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any()).Return(mgmtnetwork.VirtualNetwork{
-					ID:   to.StringPtr("/subscriptions/id"),
-					Type: to.StringPtr("Microsoft.Network/virtualNetworks"),
-					VirtualNetworkPropertiesFormat: &mgmtnetwork.VirtualNetworkPropertiesFormat{
-						DhcpOptions: &mgmtnetwork.DhcpOptions{
-							DNSServers: &[]string{},
-						},
-						Subnets: &[]mgmtnetwork.Subnet{
-							{
-								ID: to.StringPtr(fmt.Sprintf("/subscriptions/%s/resourceGroups/test-cluster/providers/Microsoft.Network/virtualNetworks/test-vnet/subnets/master", mockSubID)),
-								SubnetPropertiesFormat: &mgmtnetwork.SubnetPropertiesFormat{
-									RouteTable: &mgmtnetwork.RouteTable{
-										ID: to.StringPtr("/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/mockrg/providers/Microsoft.Network/routeTables/routetable1"),
-									},
-								},
-							},
-						},
-					},
-				}, nil)
-
-				routeTables.EXPECT().Get(gomock.Any(), "mockrg", "routetable1", "").Return(mgmtnetwork.RouteTable{
-					ID:   to.StringPtr("/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/mockrg/providers/Microsoft.Network/routeTables/routetable1"),
-					Name: to.StringPtr("routetable1"),
-				}, nil)
-
-				resources.EXPECT().GetByID(gomock.Any(), "/subscriptions/id", azureclient.APIVersion("Microsoft.Storage")).Return(mgmtfeatures.GenericResource{
-					Name:     to.StringPtr("storage"),
-					ID:       to.StringPtr("/subscriptions/id"),
-					Type:     to.StringPtr("Microsoft.Storage/storageAccounts"),
-					Location: to.StringPtr("eastus"),
-				}, nil)
-
-				virtualMachines.EXPECT().Get(gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any()).Return(mgmtcompute.VirtualMachine{
-					ID:   to.StringPtr("/subscriptions/id"),
-					Type: to.StringPtr("Microsoft.Compute/virtualMachines"),
-					VirtualMachineProperties: &mgmtcompute.VirtualMachineProperties{
-						ProvisioningState: to.StringPtr("Succeeded"),
-					},
-				}, nil)
+			mocks: func(tt *test, resources *mock_features.MockResourcesClient, virtualMachines *mock_compute.MockVirtualMachinesClient, virtualNetworks *mock_network.MockVirtualNetworksClient, routeTables *mock_network.MockRouteTablesClient, diskEncryptionSets *mock_compute.MockDiskEncryptionSetsClient) {
+				defaultListByResourceGroup(resources)
+				defaultVirtualMachines(virtualMachines)
```

Comment:
From what I see `defaultListByResourceGroup`  and `defaultVirtualMachines` are present in every test case. We discussed it in our call: things that are the same in every case should got into the test body (`t.Run`). And I think they don't need own functions in that case.

Reply:
Updated as per call today.